git/remote.py

```diff
@@ -694,10 +715,26 @@ def push(self, refspec=None, progress=None, **kwargs):
 
         :param refspec: see 'fetch' method
         :param progress:
-            Instance of type RemoteProgress allowing the caller to receive
-            progress information until the method returns.
             If None, progress information will be discarded
 
+            No further progress information is returned after push returns.
+
+            A function (callable) that is called with the progress infomation:
+
+            progress( op_code, cur_count, max_count=None, message='' )
+
+            op_code is a bit mask of values defined in git.RemoteProgress
```

Comment:
I'd rather remove the RemoteProgress from public view. See comment in next section.